### PR TITLE
Support API key as an input

### PR DESCRIPTION
**Pull request by andyzg**
Enable the user to add an optional custom `apiKey` to the `Editor`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@codeium/react-code-editor",
-  "version": "1.0.0",
+  "version": "1.0.4",
```
Comment (khou22) [inline package.json]: Why the skip?

File: src/components/CodeiumEditor/CompletionProvider.ts [hunk 2]
```diff
@@ -59,14 +59,16 @@ export class MonacoCompletionProvider {
...
+    readonly apiKey?: string | undefined
+
   ) {
     this.metadata = new Metadata({
       ideName: getBrowserVersion() ?? "unknown",
       ideVersion: getCurrentURL() ?? "unknown",
       extensionName: "@codeium/react-code-editor",
       extensionVersion: getPackageVersion() ?? "unknown",
-      apiKey: "d49954eb-cfba-4992-980f-d8fb37f0e942", // PLAYGROUND_API_KEY
+      apiKey: apiKey || "d49954eb-cfba-4992-980f-d8fb37f0e942",
```
Comment (khou22) [inline src/components/CodeiumEditor/CompletionProvider.ts]: I've been using `??` in these cases recently, doesn't really matter so up to you

File: src/components/CodeiumEditor/CompletionProvider.ts [hunk 3]
```diff
@@ -59,14 +59,16 @@ export class MonacoCompletionProvider {
...
+    readonly setMessage: (message: string) => void,
+    readonly apiKey?: string | undefined
+
   ) {
     this.metadata = new Metadata({
       ideName: getBrowserVersion() ?? "unknown",
       ideVersion: getCurrentURL() ?? "unknown",
       extensionName: "@codeium/react-code-editor",
       extensionVersion: getPackageVersion() ?? "unknown",
-      apiKey: "d49954eb-cfba-4992-980f-d8fb37f0e942", // PLAYGROUND_API_KEY
```
Comment (khou22) [inline src/components/CodeiumEditor/CompletionProvider.ts:69]: Add a constant at the top of the file for the playground key

**Review by khou22 — APPROVED** [review]
LGTM

File: package.json (hunk 1 above)
Comment (andyzg) [reply package.json]: don't think the ones in between were version controlled. Current version is 1.0.4, will update to 1.0.5 and then merge and publish

**Pull request merged**